[5.2] Creating new project and running `npx astro add tailwind` fails

### Astro Info

```block
Astro                    v5.2.0
Node                     v20.12.1
System                   macOS (arm64)
Package Manager          npm
Output                   static
Adapter                  none
Integrations             none
```

### If this issue only occurs in one browser, which browser is a problem?

_No response_

### Describe the Bug

After creating a new Astro 5.2 project:

`npm create astro@latest`

running:

`npx astro add tailwind`

fails:

```
❯ npx astro add tailwind
✔ Resolving packages...

  Astro will run the following command:
  If you skip this step, you can always run it yourself later

 ╭───────────────────────────────────────────────────────────────────────────────────╮
 │ npm install @astrojs/tailwind@^6.0.0 @tailwindcss/vite@^4.0.1 tailwindcss@^4.0.1  │
 ╰───────────────────────────────────────────────────────────────────────────────────╯

✔ Continue? … yes
✖️ Installing dependencies...

 The command `npm install @astrojs/tailwind@^6.0.0 @tailwindcss/vite@^4.0.1 tailwindcss@^4.0.1` exited with code 1 

Astro could not update your astro.config.js file safely.
Reason: Unable to install dependencies

You will need to add these integration(s) manually.
Documentation: https://docs.astro.build/en/guides/integrations-guide/
  Location:
    /Users/marceloverdijk/workspace/astro-52-tailwind/node_modules/astro/dist/cli/add/index.js:217:31
  Stack trace:
    at add (file:///Users/marceloverdijk/workspace/astro-52-tailwind/node_modules/astro/dist/cli/add/index.js:217:31)
    at async runCommand (file:///Users/marceloverdijk/workspace/astro-52-tailwind/node_modules/astro/dist/cli/index.js:124:7)


```


Running the `npm i` manually it gives more details:

```
❯ npm install @astrojs/tailwind@^6.0.0 @tailwindcss/vite@^4.0.1 tailwindcss@^4.0.1
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: astro-52-tailwind@0.0.1
npm ERR! Found: tailwindcss@4.0.1
npm ERR! node_modules/tailwindcss
npm ERR!   tailwindcss@"^4.0.1" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer tailwindcss@"^3.0.24" from @astrojs/tailwind@6.0.0
npm ERR! node_modules/@astrojs/tailwind
npm ERR!   @astrojs/tailwind@"^6.0.0" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! 
npm ERR! For a full report see:
npm ERR! /Users/marceloverdijk/.npm/_logs/2025-01-30T16_13_46_751Z-eresolve-report.txt

npm ERR! A complete log of this run can be found in: /Users/marceloverdijk/.npm/_logs/2025-01-30T16_13_46_751Z-debug-0.log
```

### What's the expected result?

Tailwind to be setup successfully.

### Link to Minimal Reproducible Example

*

### Participation

- [ ] I am willing to submit a pull request for this issue.

--- a/.changeset/green-sloths-switch.md
+++ b/.changeset/green-sloths-switch.md
@@ -2,4 +2,4 @@
 'astro': patch
 ---
 
-Fixes a bug that caused peer dependency errors when running "astro add tailwind"
+Fixes a bug that caused peer dependency errors when running `astro add tailwind`

--- a/.changeset/green-sloths-switch.md
+++ b/.changeset/green-sloths-switch.md
@@ -1,0 +1,5 @@
+---
+'astro': patch
+---
+
+Fixes a bug that caused peer dependency errors when running "astro add tailwind"

--- a/packages/astro/src/cli/add/index.ts
+++ b/packages/astro/src/cli/add/index.ts
@@ -40,6 +40,7 @@ interface AddOptions {
 interface IntegrationInfo {
 	id: string;
 	packageName: string;
+	integrationName: string;
 	dependencies: [name: string, version: string][];
 	type: 'integration' | 'adapter';
 }
@@ -283,7 +284,7 @@ export async function add(names: string[], { flags }: AddOptions) {
 						'SKIP_FORMAT',
 						`\n  ${magenta(
 							`Check our deployment docs for ${bold(
-								integration.packageName,
+								integration.integrationName,
 							)} to update your "adapter" config.`,
 						)}`,
 					);
@@ -349,7 +350,9 @@ export async function add(names: string[], { flags }: AddOptions) {
 		case UpdateResult.failure:
 		case UpdateResult.updated:
 		case undefined: {
-			const list = integrations.map((integration) => `  - ${integration.packageName}`).join('\n');
+			const list = integrations
+				.map((integration) => `  - ${integration.integrationName}`)
+				.join('\n');
 			logger.info(
 				'SKIP_FORMAT',
 				msg.success(
@@ -618,8 +621,7 @@ async function convertIntegrationsToInstallSpecifiers(
 	integrations: IntegrationInfo[],
 ): Promise<string[]> {
 	const ranges: Record<string, string> = {};
-	for (let { packageName, dependencies } of integrations) {
-		ranges[packageName] = '*';
+	for (let { dependencies } of integrations) {
 		for (const [name, range] of dependencies) {
 			ranges[name] = range;
 		}
@@ -790,7 +792,7 @@ async function validateIntegrations(integrations: string[]): Promise<Integration
 
 				const resolvedScope = pkgType === 'first-party' ? '@astrojs' : scope;
 				const packageName = `${resolvedScope ? `${resolvedScope}/` : ''}${name}`;
-
+				let integrationName = packageName;
 				let dependencies: IntegrationInfo['dependencies'] = [
 					[pkgJson['name'], `^${pkgJson['version']}`],
 				];
@@ -823,13 +825,19 @@ async function validateIntegrations(integrations: string[]): Promise<Integration
 				}
 
 				if (integration === 'tailwind') {
+					integrationName = 'tailwind';
 					dependencies = [
 						['@tailwindcss/vite', '^4.0.0'],
 						['tailwindcss', '^4.0.0'],
 					];
 				}
-
-				return { id: integration, packageName, dependencies, type: integrationType };
+				return {
+					id: integration,
+					packageName,
+					dependencies,
+					type: integrationType,
+					integrationName,
+				};
 			}),
 		);
 		spinner.success();

